### PR TITLE
feat: price impact warning

### PR DESCRIPTION
**Pull request by zzmp**
- Moves USDC price impact computation to a hook
- Fixes popover padding for more uniform popovers (and less custom styling)
- Adds a warning on the swap screen (see image)

image

### CODE REVIEW

**Comment by zzmp** [comment]
Cal has requested 0.6 opacity instead of 0.4 for loadingCss.

File: src/lib/components/Swap/Toolbar/Caption.tsx [hunk 1]
```diff
@@ -63,21 +66,21 @@ export function WrapCurrency({ loading, wrapType }: { loading: boolean; wrapType
 
 export function Trade({ trade }: { trade: InterfaceTrade<Currency, Currency, TradeType> }) {
   const [flip, setFlip] = useState(true)
-  const { inputAmount, outputAmount, executionPrice } = trade
-  const fiatValueInput = useUSDCPrice(inputAmount.currency)
-  const fiatValueOutput = useUSDCPrice(outputAmount.currency)
+  const { inputAmount: input, outputAmount: output, executionPrice } = trade
+  const { inputUSDC, outputUSDC, priceImpact } = useUSDCPriceImpact(input, output)
+  const isPriceImpactHigh = priceImpact && getPriceImpactWarning(priceImpact) === 'error'
```
Comment (zzmp) [inline src/lib/components/Swap/Toolbar/Caption.tsx:71]: This should check if `priceImpact && getPriceImpactWarning(priceImpact)`, as it should be displayed whether the warning state is `warning` _or_ `error`.

File: src/lib/hooks/useUSDCPriceImpact.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Currency, CurrencyAmount, Percent, Token } from '@uniswap/sdk-core'
+import { useUSDCValue } from 'hooks/useUSDCPrice'
+import { useMemo } from 'react'
+import { computeFiatValuePriceImpact } from 'utils/computeFiatValuePriceImpact'
+
+export default function useUSDCPriceImpact(
+  input: CurrencyAmount<Currency> | undefined,
```
Comment (ianlapham) [inline src/lib/hooks/useUSDCPriceImpact.ts:7]: Can we call this inputAmount?

File: src/lib/components/Swap/Summary/Summary.tsx [hunk 3]
```diff
@@ -42,11 +25,15 @@ function TokenValue({ input, usdc, change }: TokenValueProps) {
...
       </Row>
-      {usdc && usdcAmount && (
+      {usdc && (
         <Row justify="flex-start">
           <ThemedText.Caption color="secondary" userSelect>
-            ${formatCurrencyAmount(usdcAmount, 2, i18n.locale)}
-            {change && <Percent gain={change > 0}> {percent}</Percent>}
+            ${formatCurrencyAmount(usdc, 2, i18n.locale)}
+            {priceImpact && (
+              <ThemedText.Caption color={priceImpact?.greaterThan(0) ? 'error' : 'success'}>
```
Comment (ianlapham) [inline src/lib/components/Swap/Summary/Summary.tsx:33]: can we use `getPriceImpactWarning` for color here?

**Review by ianlapham — COMMENTED** [review]
Small changes

File: src/lib/components/Swap/Summary/Summary.tsx (hunk 3 above)
Comment (zzmp) [reply src/lib/components/Swap/Summary/Summary.tsx:33]: also documented in [notion](https://www.notion.so/uniswaplabs/0-105-price-impact-is-red-on-swap-review-screen-08a28eef9c6741fdad1cbcecfa976a2a)

**Review by JFrankfurt — APPROVED** [review]

**Review by ianlapham — APPROVED** [review]
Looks good

**Pull request merged**